src/components/ui/dashboard/coaching-session-list.tsx

```diff
@@ -0,0 +1,154 @@
+"use client";
+
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { format } from "date-fns";
+import { ArrowUpDown, CalendarPlus } from "lucide-react";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import CoachingSession from "@/components/ui/coaching-session";
+import { useAuthStore } from "@/lib/providers/auth-store-provider";
+import { useCoachingRelationshipStateStore } from "@/lib/providers/coaching-relationship-state-store-provider";
+import {
+  createCoachingSession,
+  useCoachingSessions,
+} from "@/lib/api/coaching-sessions";
+
+export default function CoachingSessionList() {
+  const { currentCoachingRelationshipId } = useCoachingRelationshipStateStore(
+    (state) => state
+  );
+  const { isCoach } = useAuthStore((state) => state);
+  const {
+    coachingSessions,
+    isLoading: isLoadingCoachingSessions,
+    isError: isErrorCoachingSessions,
+    mutate,
+  } = useCoachingSessions(currentCoachingRelationshipId);
+
+  const [sortByDate, setSortByDate] = useState(true);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [newSessionDate, setNewSessionDate] = useState("");
+
+  const handleCreateSession = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Format the date string
+    const formattedDate = format(
+      new Date(newSessionDate),
```

Comment:
I've been using ts-luxmon for working with Dates and Times because it's far more powerful and flexible in its parsing of standard date/time-stamps (e.g. ISO standards).

Do you agree that consistency is important? If so, take a look at how I use ts-luxmon in src/types/session-title.ts as an example. I also allow for one's locale to influence the formatting of the date with a system-wide config value.

Reply:
Completely agree efe7867e005c8b1b49019cd7b53dcb76d9d710a9. I blindly went with what v0.dev offered :)